framework/src/st_things/things_stack/src/common/framework/things_security_manager.c

```diff
@@ -172,7 +169,109 @@ const unsigned char OIC_SVR_DB_DOXM_PAYLOAD[] = {
 	0x65, 0xFF, 0xFF
 };
 
-FILE *server_fopen(const char *path, const char *mode)	// pkss
+#ifdef CONFIG_SUPPORT_FULL_SECURITY
+typedef enum {
+	SVR_IDLE = 0,
+	SVR_READ,
+	SVR_WRITE
+} svr_mode_t;
+
+static struct {
+	uint8_t *ptr;
+	uint8_t data[SEE_IOTIVITY_MAXSIZE];
+	uint32_t len;
+	svr_mode_t mode;
+} svr;
+
+FILE *server_fopen(const char *filename, const char *mode)
+{
+	svr.ptr = svr.data;
+	if (strcmp(mode, "r") == 0 || strcmp(mode, "rb") == 0) {
+		see_read_iotivity_storage((unsigned char *)svr.data, SEE_IOTIVITY_MAXSIZE, &svr.len);
+		svr.mode = SVR_READ;
+		if (svr.len <= 10) {
```

Comment:
What's the significance of  this conditional check with 10 ? any comment would be appreciable to understand this logic

Reply:
@pradeepns85 it was an unclear point. I changed it to distinct value.